official/keras_application_models/model_callbacks.py

```diff
@@ -0,0 +1,169 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Callbacks for Keras built-in application models."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import time
+
+import tensorflow as tf  # pylint: disable=g-bad-import-order
+
+from official.utils.logs import logger
+
+# Metrics to log after each batch and epoch
+_PER_BATCH_METRICS = {
+    "loss": "train_loss",
+    "acc": "train_accuracy",
+}
+_PER_EPOCH_METRICS = {
+    "loss": "train_loss",
+    "acc": "train_accuracy",
+    "val_loss": "loss",
+    "val_acc": "accuracy"
+}
+
+
+class ExamplesPerSecondCallback(tf.keras.callbacks.Callback):
+  """ExamplesPerSecond callback.
+
+  This callback records the average_examples_per_sec and
+  current_examples_per_sec during training.
+  """
+
+  def __init__(self, batch_size, every_n_steps=1, metric_logger=None):
+    self._batch_size = batch_size
+    self._every_n_steps = every_n_steps
+    self._logger = metric_logger or logger.BaseBenchmarkLogger()
+    self._global_step = 0
+    super(ExamplesPerSecondCallback, self).__init__()
+
+  def on_train_begin(self, logs=None):
+    self._train_time = 0
+    self._batch_times = []
+
+  def on_batch_begin(self, batch, logs=None):
+    self._time_start = time.time()
+
+  def on_batch_end(self, batch, logs=None):
+    """Log examples_per_sec.
+
+    Record the elapsed time of each batch, and log the examples_per_sec metrics
+    every_n_steps.
+    """
+    self._global_step += 1
+
+    elapsed_time = time.time() - self._time_start
+    self._train_time += elapsed_time
+    self._batch_times.append(elapsed_time)
+
+    if self._global_step % self._every_n_steps == 0:
+      average_examples_per_sec = self._batch_size * (
+          self._global_step / self._train_time)
+      self._logger.log_metric(
+          "average_examples_per_sec", average_examples_per_sec,
+          global_step=self._global_step)
+
+      # Get the total elapsed time of the last n steps
+      n_batch_time = sum(self._batch_times[-self._every_n_steps:])
+      current_examples_per_sec = self._batch_size * (
+          self._every_n_steps / n_batch_time)
+      self._logger.log_metric(
+          "current_examples_per_sec", current_examples_per_sec,
+          global_step=self._global_step)
+
+
+class LoggingMetricCallback(tf.keras.callbacks.Callback):
+  """LoggingMetric callback.
+
+  Log the predefined _PER_BATCH_METRICS after each batch, and log the predefined
+  _PER_EPOCH_METRICS after each epoch.
+  """
+
+  def __init__(self, metric_logger=None):
+    self._logger = metric_logger or logger.BaseBenchmarkLogger()
+    self._per_batch_metrics = _PER_BATCH_METRICS
```

Comment:
Should we expose them as the constructor params?

Reply:
Here we decide to use the pre-defined metrics as Karmel suggested to make the code simple.